Dictionary.java

```diff
@@ -0,0 +1,29 @@
+import java.util.ArrayList;
+
+public class Dictionary {
+	private ArrayList<String> dict;
+
+	public Dictionary(String[] dict) {
+		this.dict = new ArrayList<>();
+		for (int i = 0; i < dict.length; i++) {
+			this.dict.add(dict[i]);
+		}
+	}
+
+	public boolean removeWord(String word) {
+		return dict.remove(word);
+	}
+
+	public boolean isWord(String word) {
+		return dict.contains(word);
+	}
+
+	public boolean isPrefix(String prefix) {
+		for (String word : dict) {
+			if (word.startsWith(prefix)) {
```

Comment:
You don't need to code this; but have a think about how this could be implemented in an efficient way.